// @wxt-dev/module-vue/dist/index.mjs
import 'wxt';
import { defineWxtModule, addViteConfig, addImportPreset } from 'wxt/modules';
import vue from '@vitejs/plugin-vue';

const index = defineWxtModule({
  name: "@wxt-dev/module-vue",
  configKey: "vue",
  setup(wxt, options) {
    const { vite } = options ?? {};
    addViteConfig(wxt, ({ command }) => ({
      // @ts-ignore: Ignore vite version issues
      plugins: [vue(vite)],
      build: {
        // Fixes known issue: https://github.com/wxt-dev/wxt/pull/607
        sourcemap: false
      }
    }));
    wxt.hook("config:resolved", (wxt2) => {
      var _a;
      if (!wxt2.config.imports)
        return;
      (_a = wxt2.config.imports).addons ?? (_a.addons = {});
      if (!Array.isArray(wxt2.config.imports.addons)) {
        wxt2.config.imports.addons.vueTemplate = true;
      } else {
        wxt2.logger.warn(
          "Could not enable auto-imports in vue templates when using and array for imports.addons"
        );
      }
    });
    addImportPreset(wxt, "vue");
  }
});

export { index as default };
